public/javascripts/Progress/src/Progress.js

```diff
@@ -422,63 +422,101 @@ function Progress (_, $, c3, L, role, difficultRegionIds) {
     }
 
     function initializeSubmittedTasks(map) {
```

Comment:
Although the person who originally wrote this didn't comment it, I'm going to ask you to add comments for this function since you've now revamped it :)

Reply:
Added comments